### PR TITLE
Allow users to set jitify cache file limit via an environment variable

### PR DESCRIPTION
**Pull request by trxcllnt**
Jitify limits the file count of the kernel cache, defaulting to the number of JIT kernels to keep in memory.

Limiting to 100 files on disk leads to thrashing if running workloads that exercise all the binops (like unit tests).

This PR adds a `LIBCUDF_KERNEL_CACHE_FILE_LIMIT` environment variable so users can control the number of files Jitify should keep on disk.

### CODE REVIEW

File: cpp/src/jit/cache.cpp [hunk 1]
```diff
@@ -112,9 +114,14 @@ jitify2::ProgramCache<>& get_program_cache(jitify2::PreprocessedProgramData prep
   auto existing_cache = caches.find(preprog.name());
 
   if (existing_cache == caches.end()) {
-    auto res = caches.insert(
-      {preprog.name(),
-       std::make_unique<jitify2::ProgramCache<>>(100, preprog, nullptr, get_program_cache_dir())});
+    auto max_files = std::getenv("LIBCUDF_KERNEL_CACHE_FILE_LIMIT");
+    auto res       = caches.insert({preprog.name(),
+                              std::make_unique<jitify2::ProgramCache<>>(
+                                100,
```
Comment (devavret) [inline cpp/src/jit/cache.cpp:120]: Why not keep a lot in the memory too?

Comment (trxcllnt) [reply cpp/src/jit/cache.cpp:120]: I don't know how much memory a JIT'd kernel takes up, so I left this number unmodified.

**Review by devavret — APPROVED** [review]

**Review by davidwendt — APPROVED** [review]

File: cpp/src/jit/cache.cpp (hunk 1 above)
Comment (cwharris) [reply cpp/src/jit/cache.cpp]: updated it to cache (effectively) unlimited in-memory and on-disk kernels, as was the behavior prior to jitify2.

**Review by devavret — APPROVED** [review]

File: cpp/src/jit/cache.cpp [hunk 2]
```diff
@@ -112,9 +123,27 @@ jitify2::ProgramCache<>& get_program_cache(jitify2::PreprocessedProgramData prep
...
+    size_t kernel_limit_proc = std::numeric_limits<size_t>::max();
+    size_t kernel_limit_disk = std::numeric_limits<size_t>::max();
+
+    try_parse_numeric_env_var(kernel_limit_proc, "LIBCUDF_KERNEL_CACHE_LIMIT_PER_PROCESS");
+    try_parse_numeric_env_var(kernel_limit_disk, "LIBCUDF_KERNEL_CACHE_LIMIT_DISK");
+
+    auto cache_dir = get_program_cache_dir();
+
+    if (kernel_limit_disk == 0) {
+      // if kernel_limit_disk is zero, jitify will inherit it the value of kernel_limit_proc.
```
Comment (trxcllnt) [inline cpp/src/jit/cache.cpp:135]:
```suggestion
      // if kernel_limit_disk is zero, jitify will inherit the value of kernel_limit_proc.
```

File: cpp/src/jit/cache.cpp [hunk 3]
```diff
@@ -112,9 +123,27 @@ jitify2::ProgramCache<>& get_program_cache(jitify2::PreprocessedProgramData prep
   auto existing_cache = caches.find(preprog.name());
 
   if (existing_cache == caches.end()) {
-    auto res = caches.insert(
-      {preprog.name(),
-       std::make_unique<jitify2::ProgramCache<>>(100, preprog, nullptr, get_program_cache_dir())});
+    size_t kernel_limit_proc = std::numeric_limits<size_t>::max();
+    size_t kernel_limit_disk = std::numeric_limits<size_t>::max();
+
```
Comment (trxcllnt) [inline cpp/src/jit/cache.cpp:128]:
```suggestion
```

Comment (davidwendt) [reply cpp/src/jit/cache.cpp:128]: Also, use `std::size_t` instead of `size_t`.

**Review by cwharris — APPROVED** [review]
so code much wow :dog2:

**Review by davidwendt — APPROVED** [review]

**Comment by trxcllnt** [comment]
@gpucibot merge

**Pull request merged**